### PR TITLE
Navigation Bar

### PR DESCRIPTION
**Pull request by LHJE**
# Navigation Bar

Go get the basic skeletal structure of the webpages. API calls are not made in this PR.

## Controllers

Controllers were made for `Welcome`, `Learn More`, `Plants`, and `Sessions`, but `Sessions` will likely be irrelevant once OAuth is in place.

## POROS

Poros were created for Plant and Sensor, but we recognize these are likely only going to be used for testing the front end right now

### Application.html.erb

The `application.html.erb` was updated with fully Nav Bar logic, with links all matching what is outlined in the wireframes.  There is also logic in there so that if a user is not logged in, they cannot see the Nav Bar.  This works when you run `rails s` and look at it, but we can't write a working test for that until OAuth is in place.   

Right now we have a test written for logging a user out and being sent to the root_path, but not being able to see the Nav bar, but because we're hard-loading a user with the `allow_any_instance_of` method, we can't un-load that user to test that they have been logged out and thus cannot see the nav bar.

### Other Views

We also created all basic view pages for everything else in the wireframe, but most do not have any code in them.  They were just needed for testing the links in the nav bar.

## Other Actions

Routes were updated accordingly.  Since the Nav Bar has to be tested to work on nearly every page, probably most of the routes that will be needed were made. 

Also, there's a Welcome Index spec that's been added, because we didn't check out to another branch before making it, and it was too late to take it out.  It works though!

## Additional Notes

Currently all tests are passing with 96.6% coverage.  The lines that aren't being hit are lines that will be hit when sensors and plants pages are made, and when OAuth has been implemented, to take care of the lines not being tested in the nav_spec

### CODE REVIEW

File: app/controllers/welcome_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class WelcomeController < ApplicationController
+  def index
+    if !current_user.nil?
+      redirect_to dashboard_path
+    end
+  end
+end
```
Comment (leahriffell) [inline app/controllers/welcome_controller.rb:7]: @LHJE @dcoleman21 Is there anything on the welcome page that we do want logged in users to have access to?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,15 +1,42 @@
...
-  <body>
-    <%= yield %>
-  </body>
+</head>
+<body>
+  <% if !current_user.nil? %>
+    <nav class="navbar navbar-expand-xl bg-dark navbar-dark sticky-top" id="navbar-<%= current_user.id %>">
+      <a class="navbar-brand" href="/">Home</a>
+      <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#collapsibleNavbar">
+        <span class="navbar-toggler-icon"></span>
```
Comment (leahriffell) [inline app/views/layouts/application.html.erb:15]: @LHJE @dcoleman21 Home = Dashboard, right? If so, should we rename this nav item accordingly?

File: spec/features/navbar/navbar_spec.rb [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    click_on 'Logout'
+
+    expect(current_path).to eq(root_path)
+
+    expect(page).to_not have_link('My Gardens')
+    expect(page).to_not have_link('My Impact')
+    expect(page).to_not have_link('Learn More')
+    expect(page).to_not have_link('Profile')
+    expect(page).to_not have_link('Logout')
+  end
```
Comment (leahriffell) [inline spec/features/navbar/navbar_spec.rb:209]: Thank you for all of this thorough testing!

**Review by leahriffell — CHANGES_REQUESTED** [review]
Good work y'all! Just left a few q's.

**Review by leahriffell — APPROVED** [review]

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (LHJE) [reply app/views/layouts/application.html.erb:15]: Good idea!

File: app/controllers/welcome_controller.rb (hunk 1 above)
Comment (LHJE) [reply app/controllers/welcome_controller.rb:7]: Not really?  It's just sort of a "Here is the basic information about what this app is", a slide show of Dani's PDF stuff (which in and of itself is not a resource I would say, more of an enticer), a link to log in, a link to the Learn More page, which the user would also have access to via the nav bar.  But we can defo chat about that before merging.

File: config/routes.rb [hunk 4]
```diff
@@ -1,8 +1,14 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  root to: 'welcome#index'
   get '/privacy', to: 'privacy#index'
-  
-  get '/dashboard', to: 'dashboard#show'
-  
-  resources :gardens, expect: :index
+  get "/users", to: "users#show"
+  resources :user, only: [:show, :update, :destroy]
```
Comment (hashmaster3k) [inline config/routes.rb:5]: Possibly duplicate of line 3 and 4?

File: config/routes.rb [hunk 5]
```diff
@@ -1,8 +1,14 @@
...
+  get "/dashboard", to: "dashboard#show"
+  resources :dashboard, only: [:show]
```
Comment (hashmaster3k) [inline config/routes.rb:7]: Same here.

File: config/routes.rb [hunk 6]
```diff
@@ -1,8 +1,14 @@
...
+  resources :gardens, except: [:index]
+  resources :plants, except: [:index]
+  resources :sensors, except: [:index]
+  get "/learn_more", to: "learn_more#show"
+  resources :learn_more, only: [:show]
```
Comment (hashmaster3k) [inline config/routes.rb:12]: More duplicate

**Review by hashmaster3k — APPROVED** [review]
Great work @dcoleman21 and @LHJE ! Other than some duplicate routes, work with @IamNorma on deciding what controller name we want for the 'home page' or 'welcome page'.

File: config/routes.rb (hunk 6 above)
Comment (LHJE) [reply config/routes.rb]: Good find on all these.  I've updated the routes so there should be no more duplicates

File: config/routes.rb [hunk 7]
```diff
@@ -1,8 +1,11 @@
...
-  get '/dashboard', to: 'dashboard#show'
-  
-  resources :gardens, expect: :index
+  resources :users, except: [:index]
+  get "/dashboard", to: "dashboard#show"
+  resources :gardens, except: [:index]
+  resources :plants, except: [:index]
+  resources :sensors, except: [:index]
+  get "/learn_more", to: "learn_more#show"
+  get "/logout", to: "sessions#destroy"
```
Comment (hashmaster3k) [inline config/routes.rb:10]: Thanks for fixing those!

**Pull request merged**